### DISCUSSION TITLE
Exposure Control / Spot Meter

### DISCUSSION THREAD

**Discussion by cornemrc** [gopro/OpenGoPro#48] [Q&A] [answered]
Hello,
is there any option to control the exposure (previously known as spot meter function)? 

One of the most important problems with the newer GoPro Cams is the unavailable option to keep the exposure locked in the newer cameras after turning them off and on again (it was possible with GoPro Hero 3 for example). 

Once the cameras are in place I am unable to activate this new exposure setting because I cannot reach its back side. So it would be perfect to have a workaround by this API at least.

For a low energy consumption the Bluetooth LE would be the only option, the standard App needs too much power over Wifi.

Within this Bluetooth LE documentation (https://gopro.github.io/OpenGoPro/ble) I have found this which looks promising:

65 | Exposure select type | Liveview Exposure Select Mode | integer | 0: Disabled1: Auto2: ISO Lock3: Hemisphere
66 | Exposure select x | Liveview Exposure Select: y-coordinate (percent) | percent | 0-100
67 | Exposure select y | Liveview Exposure Select: y-coordinate (percent) | percent | 0-100

But as far as I can see those are "status codes", so some kind of "read only"?

Is there any way to do this already or is this planned?

**Comment by tcamise-gpsw** [comment, marked as answer]
Hello. Your interpretation of the BLE spec is correct: these are currently (read-only) statuses. No, there is currently no way to control the exposure.

I've created https://github.com/gopro/OpenGoPro/issues/51 from this and will add it to the internal list of feature requests. I will update this ticket when / if more information is available.

**Reply by imayka** [reply]
Hello. Is there any way to set exposure compensation using ble/wifi api? I want to use this function to make series of  photo of raw images with different exposure to combine them later into HDR (on PC or smartphone). Also didn't find possibility to set output format (jpeg/raw), is it available in api?

**Reply by imayka** [reply]
For raw output  i have a workaround (can set manually),  but for exposure compensation for my opinion it's just must have feature (to escape  touching of camera between shots). I hope to see this feature in new release..

**Reply by KonradIT** [reply]
You can control EV comp using the commands I outlined in my unofficial docs.

For Hero8 and earlier cams:

For photo mode, try with these: https://github.com/KonradIT/goprowifihack/blob/master/HERO7/HERO7-Commands.md#ev-steps-1 since those are wifi-commands.

For Hero9 and newer:

Over BLE:

`ble-send 34 03760108` would set the current mode to -2.0 (118 -> 0x76 hex)

**Reply by imayka** [reply]
> You can control EV comp using the commands I outlined in my unofficial docs.
> 
> For Hero8 and earlier cams:
> 
> For photo mode, try with these: https://github.com/KonradIT/goprowifihack/blob/master/HERO7/HERO7-Commands.md#ev-steps-1 since those are wifi-commands.
> 
> For Hero9 and newer:
> 
> Over BLE:
> 
> `ble-send 34 03760108` would set the current mode to -2.0 (118 -> 0x76 hex)

Thanks a lot! Could you also explain how to gat another values (-2...+2)? For me 118 (0x76) is just a magic number, can't find mention in your repo. What means each number in this command?  (for 34 SetSetting it's clear)
